wkcuber/api/properties.py

```diff
@@ -0,0 +1,220 @@
+
+
+@attr.define
+class LayerViewConfiguration:
+    """
+    Stores information on how the dataset is shown in webknossos by default.
+    """
+
+    color: Optional[Tuple[int, int, int]] = None
+    alpha: Optional[float] = None
+    intensity_range: Optional[Tuple[float, float]] = None
+    min: Optional[float] = None  # pylint: disable=redefined-builtin
+    max: Optional[float] = None  # pylint: disable=redefined-builtin
+    is_disabled: Optional[bool] = None
+    is_inverted: Optional[bool] = None
+    is_in_edit_mode: Optional[bool] = None
+
+
+# --- Property --------------------
+
+
+@attr.define
+class MagViewProperties:
+    resolution: Union[int, Mag]
+    cube_length: int
+
+
+@attr.define
+class LayerProperties:
+    name: str
+    category: str
+    bounding_box: BoundingBox
+    element_class: str
+    wkw_resolutions: List[MagViewProperties]
+    data_format: str
+    num_channels: Optional[int] = None
+    default_view_configuration: Optional[LayerViewConfiguration] = None
+
+
+@attr.define
+class SegmentationLayerProperties(LayerProperties):
+    largest_segment_id: int = -1
+    mappings: List[str] = []
+
+
+@attr.define
+class DatasetProperties:
+    id: Dict[str, str]
+    scale: Tuple[float, float, float]
+    data_layers: List[Union[SegmentationLayerProperties, LayerProperties]]
+    default_view_configuration: Optional[DatasetViewConfiguration] = None
+
+
+# --- Converter --------------------
+
+dataset_converter = cattr.Converter()
+
+# register (un-)structure hooks for non-attr-classes
+bbox_to_wkw: Callable[[BoundingBox], dict] = lambda o: o.as_wkw()
+dataset_converter.register_unstructure_hook(BoundingBox, bbox_to_wkw)
+dataset_converter.register_structure_hook(
+    BoundingBox, lambda d, _: BoundingBox.from_wkw(d)
+)
+
+mag_to_array: Callable[[Mag], List[int]] = lambda o: o.to_array()
+dataset_converter.register_unstructure_hook(Mag, mag_to_array)
+dataset_converter.register_structure_hook(Mag, lambda d, _: Mag(d))
+
+# register (un-)structure hooks for attr-classes
+for cls in [
+    DatasetProperties,
+    LayerProperties,
+    SegmentationLayerProperties,
+    MagViewProperties,
+    DatasetViewConfiguration,
+    LayerViewConfiguration,
+]:
+    dataset_converter.register_unstructure_hook(
+        cls,
+        make_dict_unstructure_fn(
+            cls,
+            dataset_converter,
+            **{
+                a.name: override(
+                    omit_if_default=True, rename=_snake_to_camel_case(a.name)
+                )
+                for a in attr.fields(cls)  # pylint: disable=not-an-iterable
+            },
+        ),
+    )
+    dataset_converter.register_structure_hook(
+        cls,
+        make_dict_structure_fn(
+            cls,
+            dataset_converter,
+            **{
+                a.name: override(rename=_snake_to_camel_case(a.name))
+                for a in attr.fields(cls)  # pylint: disable=not-an-iterable
+            },
```

Comment:
Wow, this is quite complex, but also super elegant :heart: Please add a comment though (maybe directly in the new function, as explained in the other comment)